### DISCUSSION TITLE
How do i get multiple keyboardmemory

### DISCUSSION THREAD

**Discussion by DungnvKaopiz** [MathKeyboardEngine/MathKeyboardEngine#4] [Q&A] [answered]
i have a keyboard like this:
![image](https://user-images.githubusercontent.com/85537564/205597485-27eafa98-59d2-4b17-bf2f-616bdfee9b7d.png)

i'm dealing with multiple inputs boxes like this 
![image](https://user-images.githubusercontent.com/85537564/205595203-a4ae1284-2e95-46ff-a1a6-6c0b751e80a1.png)
when i click a box i get the box element by it's id and pass the editModeLatex to typesetLatexInOutputBox() like your exemple code and have two box with diffrent results:
![image](https://user-images.githubusercontent.com/85537564/205596319-a0b3db83-2a06-413a-94d9-65431eda6cd9.png)

but the problem here is when i open the keyboard it still get the old keyboardMemory, i want it to get data according to the box being clicked, for exemple when i click box 1, it'll show 14333 and when i click box 2 i'll have 1433399000 show on the keyboard input box

**Comment by SymboLinker** [comment, marked as answer]
Hi!

Thanks for your question.

I've created a branch https://github.com/MathKeyboardEngine/MathKeyboardEngine/tree/multipleTypesetLatexBoxes
with this commit https://github.com/MathKeyboardEngine/MathKeyboardEngine/commit/91fcc38d42283dc2a9284092fe81fd11eed0a857
that shows how to use multiple keyboardMemories. I might cherry pick it to the master branch this week, but I would like to test a few things a bit more before I do that. I hope that this helps you to continue developing!

Kind regards,
SymboLinker

**Reply by DungnvKaopiz** [reply]
Thanks for the reply,  i'll be sure to test it tomorrow and give some feedback !

**Reply by DungnvKaopiz** [reply]
I'm trying to use your code but i can't seem to fingure it out, here's the detail of what im doing:
- First I have 2 boxes (box1, box2) and a keyboard has a keyboardBox built within.
- When i click the 1 of the box (box1), i open the keyboard and type something -> the text will appear as i type in the keyboardBox -> then i hit a save button -> it shows the text to the box1. Same thing when i click the box2.
- But the problem is when i click the box1, the keyboardBox still have the old keyboardMemory so it dosen't show the text from box1 on the keyboardBox. 
- So i need some how to store the keyboardMemory of the box1 or box2 and add it to the keyboardBox when keyboard is open

**Comment by DungnvKaopiz** [comment]
ok, i use your suggested to make an array to store keyboardMemory and did it, thanks for the answer
